### DISCUSSION TITLE
Remaping mouse buttons

### DISCUSSION THREAD

**Discussion by acvar1** [labwc/labwc#955] [Q&A]
I have a mouse that uses a non standard button arrangement.  I need to remap my buttons and I know how to do that, but I do not know what to remap them to.

I need to remap BTN_EXTRA to whatever labwc is looking for "scroll right", and BTN_SIDE to whatever labwc needs for "scroll left".  I looked through the source, but could not find what I was looking for.

**Comment by Consolatis** [comment]
We rely on wlroots to get notified about input events.
The libinput backend regarding scrolling (`axis events`) is here: https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/0.16/backend/libinput/pointer.c#L78

That might not be very helpful either so I assume there is further research in libinput required.
